bin/test-changed.sh

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env bash
+
+#
+# Building and testing all the exercises on CI takes a long time
+# and can cause CI to time out.
+# Rather than testing them all we can just test the ones that have
+# changed on the current branch against the master branch.
+#
+
+XPURESCRIPT=$(dirname "$BASH_SOURCE")
+XPURESCRIPT=$(readlink -f "$XPURESCRIPT/..")
+cd "$XPURESCRIPT"
+
+declare -i TEST_RESULT=0
+FAILED_EXERCISES=''
+
+cd exercises
+
+FILES_CHANGED=$(
+  git diff --name-status master..HEAD |
```

Comment:
Yup. I couldn't work out if there was a way to get the SHA of the previous passing commit on master. Possibly could use their API.